.crews.php

```diff
@@ -189,25 +189,26 @@ function main () {
 
     require_once ".db_config.php";
 
-    parse_str(file_get_contents("php://input"), $post);
+    parse_str(file_get_contents("php://input"), $_POST);
 
-    if(!isset($post['session_id']) || (!isset($post['view_date']) && !isset($post['confirmcrew']) && !isset($post['clearcrew']))) {
+    if(!isset($_POST['session_id']) || (!isset($_POST['view_date']) && !isset($_POST['confirmcrew']) && !isset($_POST['clearcrew']))) {
         header('Bad Request', true, 400);
         echo 'Bad Request';
         exit;
     }
 
-    session_id($post['session_id']);
-    session_start();
-
-    if(!isset($_SESSION['username'])) {
+    include ".get_user_metadata.php";
+    $user = getUser($_POST['session_id']);
+    $user = json_decode($user);
+    $username = $user->username;
+    $response['username'] = $username;
+    echo json_encode($response);
+    if(!isset($username)) {
         header('Unauthorized', true, 409);
```

Comment:
Still not sure if this is the write code. A 409 is `conflict`, but this should probably be a 403.

Reply:
I stand corrected per @MasterOdin, this should be a 401